### ISSUE TITLE
Add the ability to do etcd snapshot to a pipe or a named pipe as a pseudo file

### ISSUE THREAD

**Issue by Ais8Ooz8** [etcd-io/etcd#16242]
### What would you like to be added?

An additional feature is needed within the ```etcdctl snapshot```. For example, ```save``` command saves to a file, ```pipe``` command sends to stdout.

### Why is this needed?

For security reasons, it may be preferable not to save etcd data to disk, but only to keep it in memory and pass it to a subsequent command.

**Comment by sauerburger** [comment]
I agree with @jmhbnz . It would be beneficial to get the snapshot as a stream in memory and encrypt the content before it goes to a persistent disk.

**Comment by evgkrsk** [comment]
It is not very convenient to write snapshot to disk just to get compressed version.

**Comment by arichtman** [comment]
FWIW I took a crack at using the process's stdout file descriptor but it seems not to work since etcd wants to create the snapshot in parts. Symlinking the FD into another directory allowed the snapshot to succeed but it just overwrote the symlink.

<img width="1117" height="120" alt="Image" src="https://github.com/user-attachments/assets/46a067fb-25e1-49a5-a78a-ca0164021356" />
